code/modules/projectiles/projectile.dm

```diff
@@ -258,6 +276,17 @@
 		error = dist_x/2 - dist_y
 	else
 		error = dist_y/2 - dist_x
+
+	target_angle = round(Get_Angle(starting,target))
+
+	if(linear_movement)
+		//If the icon has not been added yet
+		if( !("[icon_state][target_angle]" in bullet_master) )
+			var/icon/I = new(icon,"[icon_state]_pixel") //Generate it.
+			I.Turn(target_angle+45)

```

Comment:
I tried replacing I.Turn(target_angle+45) with I.transform = turn(I.transform, target_angle+45) after having made I an /image but then the projectiles wouldn't appear properly rotated anymore. I'll keep this for now.